packages/vite/src/client/overlay.ts

```diff
@@ -157,7 +157,7 @@ export class ErrorOverlay extends HTMLElement {
       el.textContent = text
     } else {
       let curIndex = 0
-      let match
+      let match: RegExpExecArray | null
```

Comment:
What is the rationale behind using `null` for some of these and `undefined` for others?

Reply:
`someRE.exec` returns `RegExpExecArray | null` and therefor `match` needs to be of this typedef

Reply:
Some other vars need to be `undefined` cause they are possibly access before assigned
And some other vars are first accessed after they are defined, so they don't need the `undefined`